File: opencv/Task1_basic_operates/code/Color Conversion.py
import cv2
import numpy as np

# 读取图像
img = cv2.imread("D:/opencv_test/all.jpg")

# 转换为灰度图并保存
gray_img = cv2.cvtColor(img, cv2.COLOR_BGR2GRAY)
cv2.imwrite("all_gray.jpg", gray_img)

# 转换为HSV图并保存
hsv_img = cv2.cvtColor(img, cv2.COLOR_BGR2HSV)
cv2.imwrite("all_hsv.jpg", hsv_img)

# 转换为Lab图并保存
lab_img = cv2.cvtColor(img, cv2.COLOR_BGR2Lab)
cv2.imwrite("all_lab.jpg", lab_img)

File: EmbemdedLinux/Task4_OpenCV_Cpp/Code/Color Conversion.cpp
#include <opencv2/opencv.hpp>
#include <iostream>

int main() {
   
    cv::Mat img = cv::imread("D:/opencv_test/all.jpg");
    if (img.empty()) {
        std::cout << "Unable to read picture" << std::endl;
        return -1;
    }

    
    cv::Mat gray_img;
    cv::cvtColor(img, gray_img, cv::COLOR_BGR2GRAY);
    if (!cv::imwrite("all_gray.jpg", gray_img)) {
        std::cout << " Unable to save grey release" << std::endl;
        return -1;
    }

   
    cv::Mat hsv_img;
    cv::cvtColor(img, hsv_img, cv::COLOR_BGR2HSV);
    if (!cv::imwrite("all_hsv.jpg", hsv_img)) {
        std::cout << "Unable to save HSV release" << std::endl;
        return -1;
    }


    cv::Mat lab_img;
    cv::cvtColor(img, lab_img, cv::COLOR_BGR2Lab);
    if (!cv::imwrite("all_lab.jpg", lab_img)) {
        std::cout << "Unable to save Lab release" << std::endl;
        return -1;
    }

    std::cout << "The picture is saved." << std::endl;
    return 0;
}